src/customer/strategies/bolt/bolt-customer-strategy.ts

```diff
@@ -0,0 +1,119 @@
+        const { methodId } = options;
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+        if (!paymentMethod || !paymentMethod.initializationData.publishableKey) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { developerConfig, publishableKey } = paymentMethod.initializationData;
+
+        return this._boltScriptLoader.load(publishableKey, paymentMethod.config.testMode, developerConfig)
+            .then(boltCheckout => this._boltClient = boltCheckout)
+            .then(() => this._store.getState());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._boltClient = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    continueAsGuest(credentials: GuestCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this.openCustomBoltCheckoutWidget(credentials.email, methodId)
+            .then(() => this._store.dispatch(this._billingAddressActionCreator.continueAsGuest(credentials, options)))
+            .then(() => this._store.getState());
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.signInCustomer(credentials, options))
+            .then(() => this.openCustomBoltCheckoutWidget(credentials.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._customerActionCreator.signOutCustomer(options));
+    }
+
+    signUp(customerAccount: CustomerAccountRequestBody, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.createCustomer(customerAccount, options))
+            .then(() => this.openCustomBoltCheckoutWidget(customerAccount.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    private openCustomBoltCheckoutWidget(email: string, methodId: string): Promise<InternalCheckoutSelectors> {
+        return new Promise(resolve => {
```

Comment:
I don't think you need to manually construct a `Promise` as `openCheckout` already returns one. So you can just chain to it and return a Promise that way.

Reply:
I will check this one

Reply:
I will change the method to make it more readable, but I need to keep `new Promise(resolve => ...)` to make all the code that left run only if the modal was closed. 

If Bolt Checkout modal was closed, it's ok, we will run default customer flow. But if shopper will continue with Bolt Checkout Modal, he/she will be redirected to Success page in the end of the flow.